patches/lua-tree.patch

```diff
@@ -52,6 +58,11 @@ schema/typedefs.lua:
 - use go.re2::match instead of string.match
 - fix tag validation; switch from decimal ASCII code to hex
 - fix validate_path regex due to go.re2::match
+- No-oping `kong.tools.utils.yield()` function.
+- Removing `time_ns()` function, as it requires the FFI interface & is currently unused by goks.
+- Adding back the `paths` typedef as it was removed in 3.0, as we're wanting to keep it around for use in goks.
```

Comment:
now i'm confused, was `paths` removed from Kong?

(nit, "we're wanting" sounds unnatural... at least to non-native ears)

Reply:
Yes, `typedefs.paths` was removed from Kong 3.0 in favor of `typedefs.router_paths`.

Regarding the grammatical comment, "we're wanting" is perfectly valid English, as "we're" is a contraction for "we are". To your point, English will always be English with its weird oddities. "we're" & "were" may or not be pronounced the same way depending on the accent, but the typical phonetic spelling of "we're" is "weer" & we're" is "wuhr".

Reply:
on paths, as defined in Kong:
- pre-3.0:
  - path : string starting with `/`; used on services and upstreams.
  - paths: array of "path" elements; used on routes.
- 3.0 and on:
  - path : string starting with `/`; used on services and upstreams.
  - router_path : string starting with `/` or `~/`
  - router_paths: array or "router_path" elements; used on routes.


(on the grammatical thing, of course i do know about contractions (took a while, but i've grown used to the british style :rofl: ) it's the gerund form that seems a bit out of phase with the indefinite tense typical in docs)

Reply:
> so my confusion was because the plural form was removed(/replaced), but the singular is still there. and yes, since we have different backwards compatibility requirements we should keep it.

Exactly, that's what I noticed as well. I'm unsure why the gateway team decided to keep `typedefs.path` around, but perhaps there is a reason. Apologize for the confusion.

> it's the gerund form that seems a bit out of phase with the indefinite tense typical in docs)

Ah, I see, I thought you were commenting on how it sounds vs. the tense. Yes, you're right, the verb form there wasn't grammatically proper. I've re-phrased it.